gabarit/template_nlp/nlp_project/package_name/models_training/model_huggingface.py

```diff
@@ -0,0 +1,741 @@
+            # Copies
+            for src, dst in zip(src_files, dst_files):
+                try:
+                    shutil.copyfile(src, dst)
+                except Exception as e:
+                    self.logger.error(f"Impossible to copy {src} to {dst}")
+                    self.logger.error("We still continue ...")
+                    self.logger.error(repr(e))
+
+        ##############################################
+        # Prepare x_train, x_valid, y_train & y_valid
+        # Also extract list of classes
+        ##############################################
+
+        # If not multilabel, transform y_train as dummies (should already be the case for multi-labels)
+        if not self.multi_label:
+            # If len(array.shape)==2, we flatten the array if the second dimension is useless
+            if isinstance(y_train, np.ndarray) and len(y_train.shape) == 2 and y_train.shape[1] == 1:
+                y_train = np.ravel(y_train)
+            if isinstance(y_valid, np.ndarray) and len(y_valid.shape) == 2 and y_valid.shape[1] == 1:
+                y_valid = np.ravel(y_valid)
+            # Transformation dummies
+            y_train_dummies = pd.get_dummies(y_train)
+            y_valid_dummies = pd.get_dummies(y_valid) if y_valid is not None else None
+            # Important : get_dummies reorder the columns in alphabetical order
+            # Thus, there is no problem if we fit again on a new dataframe with shuffled data
+            list_classes = list(y_train_dummies.columns)
+            # FIX: valid test might miss some classes, hence we need to add them back to y_valid_dummies
+            if y_valid_dummies is not None and y_train_dummies.shape[1] != y_valid_dummies.shape[1]:
+                for cl in list_classes:
+                    # Add missing columns
+                    if cl not in y_valid_dummies.columns:
+                        y_valid_dummies[cl] = 0
+                y_valid_dummies = y_valid_dummies[list_classes]  # Reorder
+        # Else keep it as it is
+        else:
+            y_train_dummies = y_train
+            y_valid_dummies = y_valid
+            if hasattr(y_train_dummies, 'columns'):
+                list_classes = list(y_train_dummies.columns)
+            else:
+                self.logger.warning(
+                    "Can't read the name of the columns of y_train -> inverse transformation won't be possible"
+                )
+                # We still create a list of classes in order to be compatible with other functions
+                list_classes = [str(_) for _ in range(pd.DataFrame(y_train_dummies).shape[1])]
+
+        # Set dict_classes based on list classes
+        dict_classes = {i: col for i, col in enumerate(list_classes)}
+
+        # Validate classes if already trained, else set them
+        if self.trained:
+            assert self.list_classes == list_classes, \
```

Comment:
`assert` statements are removed at compilation time. see [bandit explanations](https://bandit.readthedocs.io/en/latest/plugins/b101_assert_used.html) so we should not rely on them

Reply:
We have other assertions in the project. I will create an issue to fix them all together. Leaving this like this for this PR.

Reply:
Issue : https://github.com/OSS-Pole-Emploi/gabarit/issues/100